### DISCUSSION TITLE
Skywalking Support for Quarkus native and Quarkus JVM

### DISCUSSION THREAD

**Discussion by nitin-aristocrat** [apache/skywalking#12332] [Q&A] [answered]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

Need support documentation of quarkus integration with skywalking for trace profiling

### Use case

I have tried to use skywalking with Quarkus framework. i found it, there is no as such documentation of quarkus and skywalking. Even in documentation, it talks about zipkins only.

Initally, i tried to integrate my quarkus native application. but i found java-agents do not works with Grall VM. So, i left quarkus native and moved to Quarkus JVM only. Skywalking java agent was able to show that api got hit and internally it triggers db call and http client call. but somehow logs are not coming.

Recently, after doing so much googling, i found that quarkus supports log4j2 and skywalking also supports the same so i added skywalking log4j2 dependencies in the project. My application is running, but logs are not coming in skywalking. I am not able to do any trace profiling because after adding skywalking log4j2 and api-toolkit-trace dependencies, skywalking is not showing end point clearly, it always show end point "GET:/".

Also in background skywalking logs are showing me error in interception. I have tried to remove mongo still getting interception issue with resteasy. adding the sceenshots. 
<img width="1533" alt="Screenshot 2024-06-13 at 12 53 13 PM" src="https://github.com/apache/skywalking/assets/52274152/fbd0004c-852d-447a-91f8-e804d2f58b26">

<img width="1721" alt="Screenshot 2024-06-13 at 12 51 51 PM" src="https://github.com/apache/skywalking/assets/52274152/6449fe23-580d-4a13-8eee-8fcde8695ba4">

Withspringboot, my POC for trace profiling works fine with logback and apm-toolkit-trace dependencies. 
<img width="1440" alt="Screenshot 2024-06-10 at 10 27 29 AM" src="https://github.com/apache/skywalking/assets/52274152/357b1b97-dd85-4103-887e-ebbdb9b9b51e">

Please suggest if i am doing something wrong. Also please share some documentation of skywalking with quarkus for the integration.
 

### Are you willing to submit a pull request to implement this on your own?

- [ ] Yes I am willing to submit a pull request on my own!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by wu-sheng** [comment, marked as answer]
Native is hard. There was a PoC for similar tech, it doesn't worj well

https://github.com/apache/skywalking-graalvm-distro

There's no one working on this anymore. So, no suggestion for that.

**Comment by nitin-aristocrat** [comment]
Hi Wu-Sheng, Thanks for reply.
What about plain Quarkus JVM integration with skywalking? The Errors i have shared are of Quarkus JVM errors. There are some interceptor releated exceptions.

**Reply by wu-sheng** [reply]
That may relate to mongodb plugin incompatible.
You could remove mongodb plugin from your plugins folder.
